From 937c2903736feb9ee415b92432a31bca7fbb059c Mon Sep 17 00:00:00 2001
From: Oleksii Kislinskyi <kislinskyi@google.com>
Date: Wed, 19 Mar 2025 22:06:59 +0000
Subject: [PATCH] Fixing the default values for pending intent at
 LocationPendingIntentTransport.

Bug: 388828203
Flag: EXEMPT security fix
Test: atest LocationProviderTest
(cherry picked from commit 225602a7d0fecefb42c2456067d91f895c72f190)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:53cf3399f5691aab020d44efff5dd01eaf16257b)
Merged-In: I7251bb506f94194d8ab7cef0a5fbda195ef6341a
Change-Id: I7251bb506f94194d8ab7cef0a5fbda195ef6341a
---
 .../server/location/provider/LocationProviderManager.java       | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/services/core/java/com/android/server/location/provider/LocationProviderManager.java b/services/core/java/com/android/server/location/provider/LocationProviderManager.java
index 542a29ae4172..5dd3a77287c5 100644
--- a/services/core/java/com/android/server/location/provider/LocationProviderManager.java
+++ b/services/core/java/com/android/server/location/provider/LocationProviderManager.java
@@ -283,6 +283,7 @@ public class LocationProviderManager extends
                 throws PendingIntent.CanceledException {
             BroadcastOptions options = BroadcastOptions.makeBasic();
             options.setDontSendToRestrictedApps(true);
+            options.setPendingIntentBackgroundActivityLaunchAllowed(false);
             // allows apps to start a fg service in response to a location PI
             options.setTemporaryAppAllowlist(TEMPORARY_APP_ALLOWLIST_DURATION_MS,
                     TEMPORARY_ALLOW_LIST_TYPE_FOREGROUND_SERVICE_ALLOWED,
@@ -325,6 +326,7 @@ public class LocationProviderManager extends
                 throws PendingIntent.CanceledException {
             BroadcastOptions options = BroadcastOptions.makeBasic();
             options.setDontSendToRestrictedApps(true);
+            options.setPendingIntentBackgroundActivityLaunchAllowed(false);
 
             mPendingIntent.send(mContext, 0, new Intent().putExtra(KEY_PROVIDER_ENABLED, enabled),
                     null, null, null, options.toBundle());
-- 
2.49.0.1077.gc0e912fd4c-goog

